File: 2026-08/src/day_24/prms_181925/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

string solution(vector<int> numLog) {
    string s;
    for (int i = 1; i < numLog.size(); i++) {
        int prv = numLog[i - 1];
        int cur = numLog[i];

        if (prv + 1 == cur) {
            s += 'w';
        } else if (prv - 1 == cur) {
            s += 's';
        } else if (prv + 10 == cur) {
            s += 'd';
        } else {
            s += 'a';
        }
    }

    return s;
}

File: 2026-08/src/day_24/prms_181925/Solution.py
from itertools import pairwise


def solution(numLog):
    res = []
    for prv, cur in pairwise(numLog):
        if prv + 1 == cur:
            res.append("w")
        elif prv - 1 == cur:
            res.append("s")
        elif prv + 10 == cur:
            res.append("d")
        else:
            res.append("a")

    return "".join(res)
